File: web/src/lib/activity/DescribedActivity.ts
import { Description } from "$lib/description/Description"
import { Expose, Type } from "class-transformer"

export class DescribedActivity {
    @Expose({ name: "_id" })
    readonly id: string

    readonly name: string

    readonly logoUrl: string | null

    @Type(() => Description)
    readonly descriptions: Array<Description>
}

File: backend/src/main/kotlin/ru/disenchanted/backend/domain/activity/DescribedActivity.kt
package ru.disenchanted.backend.domain.activity

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import ru.disenchanted.backend.domain.description.Description

@Serializable
data class DescribedActivity(
    @SerialName("_id")
    val id: String,

    @SerialName("name")
    val name: String,

    @SerialName("logoUrl")
    val logoUrl: String?,

    @SerialName("descriptions")
    val descriptions: List<Description>
)
